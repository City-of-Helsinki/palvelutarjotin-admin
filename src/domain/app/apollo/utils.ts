import apolloClient from './apolloClient';

export const clearApolloCache = async () => {
  apolloClient.stop();
  await Promise.all([apolloClient.resetStore(), apolloClient.clearStore()]);
};
