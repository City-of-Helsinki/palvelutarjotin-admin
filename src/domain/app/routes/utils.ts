import { ROUTES } from './constants';

export const getCmsPath = (slug?: string | null) => {
  return ROUTES.CMS_PAGE.replace('/:slug', slug ?? '');
};
