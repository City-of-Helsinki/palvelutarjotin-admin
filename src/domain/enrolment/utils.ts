import { EnrolmentFieldsFragment } from '../../generated/graphql';

export const getEnrolmentFields = (
  enrolment: EnrolmentFieldsFragment | undefined | null
) =>
  enrolment
    ? {
        extraNeeds: enrolment.studyGroup.extraNeeds,
        language:
          enrolment.person?.language || enrolment.studyGroup.person.language,
        personInfo: [
          enrolment.person?.phoneNumber,
          enrolment.person?.emailAddress,
        ].filter((e) => e),
        studyGroupPersonInfo: [
          enrolment.studyGroup.person.phoneNumber,
          enrolment.studyGroup.person.emailAddress,
        ].filter((e) => e),
      }
    : {};
