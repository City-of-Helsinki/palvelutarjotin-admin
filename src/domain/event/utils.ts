import { SUPPORT_LANGUAGES } from '../../constants';
import { EventQuery } from '../../generated/graphql';
import { Language } from '../../types';
import { EVENT_PLACEHOLDER_IMAGES } from './constants';
/**
 * Get event placeholder image url
 * @param {string} id
 * @return {string}
 */
export const getEventPlaceholderImage = (id: string): string => {
  const numbers = id.match(/\d+/g);
  const sum = numbers
    ? numbers.reduce((prev: number, cur: string) => prev + Number(cur), 0)
    : 0;
  const index = sum % 4;

  return EVENT_PLACEHOLDER_IMAGES[index];
};

/**
 * Get first availanle language of an event
 * @param {object} eventData
 * @return {string}
 */
export const getFirstAvailableLanguage = (eventData: EventQuery): Language => {
  if (eventData.event?.name.en) return SUPPORT_LANGUAGES.EN;
  if (eventData.event?.name.fi) return SUPPORT_LANGUAGES.FI;
  if (eventData.event?.name.sv) return SUPPORT_LANGUAGES.SV;

  return SUPPORT_LANGUAGES.FI;
};
