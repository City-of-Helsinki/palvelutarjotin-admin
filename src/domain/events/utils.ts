import { isFuture } from 'date-fns';
import forEach from 'lodash/forEach';

import { EventFieldsFragment } from '../../generated/graphql';

export const hasOccurrences = (event: EventFieldsFragment) => {
  return Boolean(event.pEvent?.occurrences.edges.length);
};

export const hasComingOccurrences = (event: EventFieldsFragment) => {
  let hasComingItems = false;

  forEach(event.pEvent?.occurrences.edges, (edge) => {
    if (edge?.node?.startTime && isFuture(new Date(edge?.node?.startTime))) {
      hasComingItems = true;
      return false;
    }
  });

  return Boolean(hasComingItems);
};
