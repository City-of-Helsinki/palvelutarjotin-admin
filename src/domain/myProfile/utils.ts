import {
  OrganisationNodeFieldsFragment,
  PersonFieldsFragment,
} from '../../generated/graphql';
import { MyProfileFormFields } from './myProfileForm/MyProfileForm';

export const getMyProfilePayload = (values: MyProfileFormFields) => ({
  name: values.name,
  phoneNumber: values.phoneNumber,
  emailAddress: values.emailAddress,
  organisations: values.organisations,
});

export const getSelectedOrganisation = (
  myProfile: PersonFieldsFragment,
  activeOrganisation: string | null
): OrganisationNodeFieldsFragment | null => {
  const organisations =
    myProfile.organisations.edges.map((edge) => ({
      ...(edge?.node as OrganisationNodeFieldsFragment),
    })) || [];

  const organisation = activeOrganisation
    ? organisations.find((item) => item.id === activeOrganisation)
    : null;

  return organisation || organisations[0];
};
