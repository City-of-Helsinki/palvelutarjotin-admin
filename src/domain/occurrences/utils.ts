import { EventFieldsFragment } from '../../generated/graphql';
import { formatIntoDateTime } from '../../utils/time/format';

export const getEventPublishedTime = (event: EventFieldsFragment) => {
  // TODO: use only this when it is available
  if (event.datePublished) {
    return formatIntoDateTime(new Date(event.datePublished));
  }

  if (event.startTime) {
    return formatIntoDateTime(new Date(event.startTime));
  }

  return null;
};
