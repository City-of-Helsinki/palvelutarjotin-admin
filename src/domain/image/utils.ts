import { ImageDocument, ImageQuery } from '../../generated/graphql';
import { initializeApolloClient } from '../app/apollo/apolloClient';

export const getImageName = (id: string) => {
  const apolloClient = initializeApolloClient();
  const data = apolloClient.readQuery<ImageQuery>({
    query: ImageDocument,
    variables: { id },
  });
  return data?.image?.name;
};
