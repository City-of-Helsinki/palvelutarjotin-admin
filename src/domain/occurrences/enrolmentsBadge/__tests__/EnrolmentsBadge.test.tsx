import * as React from 'react';

import { render, screen } from '../../../../utils/testUtils';
import EnrolmentBadge from '../EnrolmentsBadge';

it('matches snapshot', () => {
  const { container } = render(
    <EnrolmentBadge
      approvedSeatsCount={10}
      pendingSeatsCount={10}
      isOccurrenceFull={false}
    />
  );

  expect(container).toMatchSnapshot();
});

it('renders seat numbers', () => {
  render(
    <EnrolmentBadge
      approvedSeatsCount={10}
      pendingSeatsCount={20}
      isOccurrenceFull={false}
    />
  );

  expect(screen.queryByText('10')).toBeInTheDocument();
  expect(screen.queryByText('20')).toBeInTheDocument();
});
