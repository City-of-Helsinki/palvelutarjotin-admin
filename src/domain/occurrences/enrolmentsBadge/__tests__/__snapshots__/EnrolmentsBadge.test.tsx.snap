// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1`] = `
<div>
  <div
    class="badgeContainer"
  >
    <div
      class="approvedSeats"
    >
      <span
        aria-hidden="true"
      >
        10
      </span>
      <div
        class="srOnly"
      >
        10 hyv채ksytty
      </div>
    </div>
    <div
      class="pendingSeats"
    >
      <span
        aria-hidden="true"
      >
        10
      </span>
      <div
        class="srOnly"
      >
        10 hyv채ksym채tt채
      </div>
    </div>
  </div>
</div>
`;
